lib/presets.js

```diff
@@ -305,21 +305,59 @@ module.exports = {
      * });
      */
     cargo: (options, cwd, version, callback) => {
-      const cargoFiles = path.join(cwd, 'Cargo.*');
+      const cargoToml = path.join(cwd, 'Cargo.toml');
+      const cargoLock = path.join(cwd, 'Cargo.lock');
       const cleanedVersion = semver.clean(version);
 
       if (!cleanedVersion) {
         return callback(new Error(`Invalid version: ${version}`));
       }
 
-      replace({
-        files: cargoFiles,
-        from: /(version\s*=\s*)"(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)"/g,
-        to: '$1"' + cleanedVersion + '"'
-      }, (error) => {
+      fs.readFile(cargoToml, {
+        encoding: 'utf8'
+      }, (error, contents) => {
         if (error) {
+          if (error.code === 'ENOENT') {
+            error.message = `No such file or directory: ${cargoToml}`;
+          }
+
           return callback(error);
         }
+
+        // Capture first `name = "..."` occurrence immediately after `[package]`
+        const matches = contents.match(/\[package\](?:\n|.)*?name\s*=\s*(?:"|')(.+)(?:"|')/m);
```

Comment:
I did a bit of searching, and it [looks like](https://stackoverflow.com/a/47317639) `[^]` might be a better version of `(?:\n|.)` ?

Reply:
Nice, that will handle `\r` as well.